helix-core/src/main/java/org/apache/helix/controller/rebalancer/waged/constraints/ConstraintBasedAlgorithmFactory.java

```diff
@@ -19,22 +19,26 @@
  * under the License.
  */
 
-import java.util.ArrayList;
 import java.util.List;
 
+import com.google.common.collect.ImmutableList;
 import org.apache.helix.controller.rebalancer.waged.RebalanceAlgorithm;
-import org.apache.helix.model.ClusterConfig;
 
 public class ConstraintBasedAlgorithmFactory {
 
   // TODO: the parameter comes from cluster config, will tune how these 2 integers will change the
   // soft constraint weight model
   public static RebalanceAlgorithm getInstance() {
-    // TODO initialize constraints, depending on constraints implementations PRs
-    List<HardConstraint> hardConstraints = new ArrayList<>();
-    List<SoftConstraint> softConstraints = new ArrayList<>();
+    List<HardConstraint> hardConstraints = ImmutableList
```

Comment:
Question: is this factory meant to be static? How do we modify the constraint based algorithm after-the-fact if the user wishes to use more constraints or if we want to add more?

Reply:
@narendly  do you mean adding or updating constraints in the run-time? I don't think it would be a good idea, it's safer and easier to add more constraints or update by normal code release. If different users have different requirements for constraints, we could definitely make it configurable in the future

Reply:
This feature is not in our current project scope. And the rebalancer won't need to know this logic.